File: api/src/admin/workspaces/entity/workspace.go
package entity

// Workspace belongs to exactly one organization. OrganizationID is
// required on create and immutable afterwards — the validator in
// `workspaces/validator` enforces both.
//
// WorkspaceID is the human-readable, admin-chosen identifier (the
// workspace's analogue of applications.client_id). It is unique
// within an organization, so two organizations can both have a
// workspace with workspace_id "default". The primary key is `id`
// (uuid); `workspace_id` is for display and, eventually, URLs.
type Workspace struct {
	_              struct{} `table:"workspace"`
	ID             string   `db:"id" dbtype:"UUID" nullable:"false" json:"id"`
	WorkspaceID    string   `db:"workspace_id" dbtype:"TEXT" nullable:"false" default:"" json:"workspace_id"`
	Title          string   `db:"title" dbtype:"TEXT" nullable:"false" json:"title"`
	Description    string   `db:"description" dbtype:"TEXT" nullable:"false" default:"" json:"description"`
	OrganizationID string   `db:"organization_id" dbtype:"TEXT" nullable:"false" json:"organization_id"`
	CreatedAt      string   `db:"created_at" dbtype:"DATETIME" nullable:"true" json:"created_at"`
	IsActive       bool     `db:"is_active" dbtype:"BOOLEAN" nullable:"false" default:"true" json:"is_active"`
}

File: app/src/Components/Workspace/model/workspace.ts
import type { Schema } from '../../../config/data/mapper/mapper.ts';

export interface Workspace {
    id: string;
    // Human-readable, admin-chosen identifier (the workspace's
    // analogue of Application.clientId). Unique per organization.
    workspaceId: string;
    title: string;
    description: string;
    organizationId: string;
    createdAt: string;
    isActive: boolean;
}

export const WorkspaceSchema: Schema = {
    id: 'id',
    workspaceId: 'workspace_id',
    title: 'title',
    description: 'description',
    organizationId: 'organization_id',
    createdAt: 'created_at',
    isActive: 'is_active',
};

export interface WorkspaceRequestResponse {
    message?: Workspace[] | null,
    success?: boolean
}

export const WorkspaceResource = 'res:workspaces';

export interface WorkspaceStatsCounts {
    total: number;
    active: number;
    inactive: number;
}

export interface ApplicationBreakdown {
    id: string;
    title: string;
    user_count: number;
    top_scopes: string[];
}

export interface WorkspaceStats {
    organization_title: string;
    created_at: string;
    is_active: boolean;
    applications: WorkspaceStatsCounts;
    users: WorkspaceStatsCounts;
    applications_breakdown: ApplicationBreakdown[];
}
